AmazonChimeSDKDemo/AmazonChimeSDKDemo/CaptionsModel.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  CaptionsModel.swift
+//  AmazonChimeSDKDemo
+//
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+
+import AmazonChimeSDK
+import UIKit
+
+class CaptionsModel: NSObject {
+    var captions = [Caption]()
+    var captionIndices = [String: Int]()    // map resultId to index in captions array
+    var refreshCaptionsTableHandler: (() -> Void)?
+
+    // triggered on main thread, no synchronization needed
+    public func addTranscriptEvent(transcriptEvent: TranscriptEvent) {
+        if let status = transcriptEvent as? TranscriptionStatus {
+            let formattedEventTime = TimeStampConversion.formatTimestamp(timestamp: status.eventTimeMs)
+            let content = "Live transcription \(status.type) at \(formattedEventTime) in \(status.transcriptionRegion) with configuration: \(status.transcriptionConfiguration)"
+            let caption = Caption(speakerName: "",
+                                  isPartial: false,
+                                  content: content)
+            captions.append(caption)
+        } else if let transcript = transcriptEvent as? Transcript {
+            transcript.results.forEach { result in
+                guard let alternative = result.alternatives.first else {
```

Comment:
Why only first?

Reply:
See comments below. Basically currently one result only has transcript from the same speaker with our dual-channel streaming connection with Transcribe, all items will have the same speaker so we just read from the first. This simplifies the logic for example, whether to add spaces between words for different languages transcribed.